streaming/src/test/scala/org/apache/spark/streaming/StreamingListenerSuite.scala

```diff
@@ -140,6 +141,37 @@ class StreamingListenerSuite extends TestSuiteBase with Matchers {
     }
     true
   }
+
+  test("registering listeners via spark.streaming.extraListeners") {
+    // Test for success with zero-argument constructor and sparkConf constructor
+    val conf = new SparkConf().setMaster("local").setAppName("test")
+      .set("spark.streaming.extraListeners",
+        classOf[StreamingListenerThatAcceptsSparkConf].getName + "," +
+        classOf[ReceiverInfoCollector].getName)
+    val scc = new StreamingContext(conf, Seconds(1))
```

Comment:
Is this still the case when the context is never started?

On Fri, Jul 17, 2015 at 1:55 AM, Tathagata Das <notifications@github.com>
wrote:

> In
> streaming/src/test/scala/org/apache/spark/streaming/StreamingListenerSuite.scala
> <https://github.com/apache/spark/pull/6380#discussion_r34868938>:
>
> > @@ -140,6 +141,37 @@ class StreamingListenerSuite extends TestSuiteBase with Matchers {
> >      }
> >      true
> >    }
> > +
> > +  test("registering listeners via spark.streaming.extraListeners") {
> > +    // Test for success with zero-argument constructor and sparkConf constructor
> > +    val conf = new SparkConf().setMaster("local").setAppName("test")
> > +      .set("spark.streaming.extraListeners",
> > +        classOf[StreamingListenerThatAcceptsSparkConf].getName + "," +
> > +        classOf[ReceiverInfoCollector].getName)
> > +    val scc = new StreamingContext(conf, Seconds(1))
>
> This can lead to a leak in the StreamingContext and underlying
> SparkContext. Please put the whole in
>
> var ssc: StremaingContext = null
> try {
>   ssc = new StreamingContext...
>
> } finally {
>    if (ssc != null) {
>      ssc.stop()
>    }
> }
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/6380/files#r34868938>.
>